Neos.Neos/Classes/Fusion/Helper/NodeHelper.php

```diff
@@ -156,6 +157,17 @@ public function subgraphForNode(Node $node): ContentSubgraphInterface
         return $this->contentRepositoryRegistry->subgraphForNode($node);
     }
 
+    public function isNodeStateHidden(Node $node): bool
```

Comment:
I just stumbled upon the name of this method. Is the _state_ of the node hidden? And how is this related to the change?

Reply:
I left it close to what it is called internally but AFAIK when we don't support property('_hidden') there would be no way to figure out if a node is hidden, so I added this helper :)

Reply:
But wouldn't `isNodeHidden()` be better since the _node_ is hidden not its state.
But more confusing to me is the fact that we decided to call it `hiddenInMenu` and `disabled` IIRC.
I also wonder whether we should introduce unused code. And with SubtreeTags (#4659) we can probably just use `node.tags.contain('disabled')`, right?

Reply:
If it works that way with the tags would be neat!

I am totally happy with removing this or changing the naming, shouldn't we then internally also use disabled everywhere? I think the point with the hidden state (finder) was originally to make it easy to just know if this node should be treated as disabled/hidden regardless if it's self hidden or just by inheritance. That's what this was meant for.